src/js/reload_on_historical.js

```diff
@@ -0,0 +1,40 @@
+import { _getById } from './modules/utils.js'
+import { WebsocketClient } from './modules/websocket_client.js'
+/*
+Listens for the status of the historical poller via an event from the websocket
+client if the #historicalBusy element shows that historical data for the site
+is still loading. Reloads the page once notified that historical data is ready.
+*/
+
+window.addEventListener('load', () => {
+  if (_getById('historicalbusy') && _getById('historicalbusy').dataset.historicalbusy !== 'True') {
+    return
+  }
+  const ws = new WebsocketClient()
+  ws.subscribe('historicalStatus')
+  window.addEventListener('historicalStatus', (message) => {
+    console.log('event message:', message)
+    const isBusy = message.detail.data
+    if (!isBusy) {
+      window.location.reload()
+    }
+  })
+})
+
+// export function waitForHistoricalStatus () {
+//   return new Promise((resolve, reject) => {
+//     const historicalBusyElement = _getById('historicalbusy')
+//     if (historicalBusyElement && historicalBusyElement.dataset.historicalbusy === 'False') {
+//       resolve()
+//       return
+//     }
+//     // eslint-disable-next-line no-unused-vars
+//     const ws = new WebsocketClient('historicalStatus')
+//     window.addEventListener('historicalStatus', (message) => {
+//       const isBusy = message.data
+//       if (!isBusy) {
+//         resolve()
+//       }
+//     })
+//   })
+// }
```

Comment:
Yes. It should really. It's there as a reminder how to have this function be called asynchronously so that the page can respond via React or at least more dynamically than just reloading when the 'historicalData' event is fired.